compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.cpp

```diff
@@ -79,10 +157,28 @@ SmallVector<Type> EncodingAttr::getElementTypesArray() {
   });
 }
 
-EncodingAttr EncodingAttr::clone(AffineMap bcastMap) {
-  return get(bcastMap.getContext(), getOperandIndex(), getOpType(),
-             getElementTypes(), getUserIndexingMaps(),
-             AffineMapAttr::get(bcastMap), getRoundDimsTo(), getLayouts());
+EncodingAttr EncodingAttr::clone(AffineMap newIndexingMap) {
+  ArrayAttr userIndexingMaps = getUserIndexingMaps();
+  SmallVector<Attribute> newMaps(userIndexingMaps.begin(),
+                                 userIndexingMaps.end());
+  unsigned operandIndex = getOperandIndex().getValue().getZExtValue();
+  assert(operandIndex < newMaps.size() && "invalid operand index");
+  if (newIndexingMap) {
+    SmallVector<Attribute> maps;
+    if (auto mapForIndex =
+            llvm::dyn_cast<AffineMapAttr>(newMaps[operandIndex])) {
+      maps.push_back(AffineMapAttr::get(mapForIndex.getAffineMap()));
+    } else if (auto mapForIndex =
+                   llvm::dyn_cast<ArrayAttr>(newMaps[operandIndex])) {
+      maps.assign(mapForIndex.begin(), mapForIndex.end());
+    }
+    maps.push_back(AffineMapAttr::get(newIndexingMap));
+    newMaps[operandIndex] = ArrayAttr::get(newIndexingMap.getContext(), maps);
+  }
+  return get(newIndexingMap.getContext(), getOperandIndex(), getOpType(),
+             getElementTypes(),
+             ArrayAttr::get(newIndexingMap.getContext(), newMaps),
+             getRoundDimsTo(), getLayouts());
```

Comment:
I think we can return `this` when newIndexingMap is a NULL, so we don't pay the cost to create a new attribute.
We can replace `newIndexingMap.getContext()` with `getContext()`, which is shorter. We also don't worry if `newIndexingMap` is null or not in this statement. (The current version is problematic when the input is NULL.)